src/data_utils/other.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Returns a name that is safe for stuff such as links.
+ * 
+ * 1. Remove extra whitespace around the name
+ * 2. Replace all spaces with underscores ("`_`")
+ * 3. Converts to lowercase.
+ * 
+ * @param name The name to convert.
+ * @returns The safe name.
+ */
+export function toSafeName(name: string) {
+    return name.trim().replaceAll(" ", "_").toLowerCase();
+}
+
+
+
+/**
+ * Returns the value if the value is a blank string or an empty array, otherwise return `undefined`.
+ * 
+ * @param value The value to check.
+ * @returns The {@link value} if {@link value} is not empty, otherwise `undefined`.
+ */
+export function ensureNotBlank<T>(value: T): T | undefined {
+    let isNotBlank =
+        value !== undefined &&
+        value !== "" &&
+        (
+            Array.isArray(value) &&
+            value.length !== 0 &&
+            value.every((innerValue) => (typeof innerValue === "string") && innerValue !== "")
+        )
+
+    return isNotBlank ? value : undefined;
+}
```

Comment:
You should probably trim the string just incase there's any spaces in the value.

Reply:
changed to a completely different system where `string` and `string[]` are handled differently:
- blank strings (`""`) are now falling back to `undefined`
- blank lists `[]` fall back to `undefined` as well
- lists with empty strings (`[""]` or `["", "", ...]`) also fall back to `undefined`